pkg/forklift-api/webhooks/mutating-webhook/mutators/ovirt-cert-mutator.go

```diff
@@ -0,0 +1,79 @@
+package mutators
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/konveyor/forklift-controller/pkg/forklift-api/webhooks/util"
+	"github.com/konveyor/forklift-controller/pkg/lib/logging"
+	admissionv1 "k8s.io/api/admission/v1beta1"
+	core "k8s.io/api/core/v1"
+)
+
+var log = logging.WithName("mutator")
+
+type OvirtCertMutator struct {
+}
+
+func (mutator *OvirtCertMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
+	log.Info("secret mutator was called")
+	raw := ar.Request.Object.Raw
+	secret := &core.Secret{}
+	err := json.Unmarshal(raw, secret)
+	if err != nil {
+		log.Error(err, "mutating webhook error")
+		util.ToAdmissionResponseError(err)
+	}
+
+	certUrl := fmt.Sprint(string(secret.Data["url"]), "/ovirt-engine/services/pki-resource?resource=ca-certificate&format=X509-PEM-CA")
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	response, err := http.Get(certUrl)
```

Comment:
better to do this after checking that the secret is for an ovirt provider, no?

Reply:
sure, it won't work to retrieve the ca if the provider is not ovirt, changed.